src/app/onboarding/components/create-new-archive/create-new-archive.component.spec.ts

```diff
@@ -1,11 +1,20 @@
 import { Shallow } from 'shallow-render';
+import { NgModule } from '@angular/core';
 
 import { CreateNewArchiveComponent } from './create-new-archive.component';
 import { OnboardingModule } from '@onboarding/onboarding.module';
 
 import { ArchiveVO } from '@models/archive-vo';
 import { ApiService } from '@shared/services/api/api.service';
 
+@NgModule({
+  declarations: [], // components your module owns.
+  imports: [], // other modules your module needs.
+  providers: [ApiService], // providers available to your module.
+  bootstrap: [] // bootstrap this root component.
+})
+class DummyModule {};
```

Comment:
(This is all for the benefit of Jest and doesn't happen in Karma.)

Basically, I'm creating a module that only has the single component in it (and necessary providers). I thought that `shallow-render` would automatically mock out everything, but it doesn't quite work and as a result if I instead try to declare the component in the `OnboardingModule`, all of the components in the OnboardingModule as well as its imported other modules get dragged into the test. That means that all the components in `SharedModule` also get dragged into the test, and if any of those components has any broken or untestable code in it, then our single component's test gets broken as well.

Reply:
Aha, I think that makes some sense! Thanks.

> I thought that `shallow-render` would automatically mock out everything, but it doesn't quite work and as a result if I instead try to declare the component in the `OnboardingModule`, all of the components in the `OnboardingModule` as well as its imported other modules get dragged into the test.

That is surprising and disappointing! Because yeah, I would also expect `shallow-render` to do the right thing here, as that is pretty much its purpose. Is the problem the application's module structure? Are our modules too tightly coupled? What would we have to do to avoid needing this test module pattern? (I am not necessarily proposing that we restructure our modules as part of this work, just trying to understand the problem!)

All of that aside, can we communicate the purpose of this via a more descriptive name? Maybe instead of `DummyModule`, we can say like `TestModule`, or even `TestOnboardingModule` in this specific case?